ui/components/app/snaps/snap-ui-renderer/snap-ui-renderer.js

```diff
@@ -119,18 +35,84 @@ export const SnapUIRenderer = ({
   isCollapsable = false,
   isCollapsed = false,
   isLoading = false,
-  data,
+  isPrompt = false,
+  // This is a workaround while we have the prompt dialog type since we can't inject the SnapUIRenderer in the template renderer.
+  inputValue,
+  onInputChange,
+  placeholder,
   onClick,
   boxProps,
+  interfaceId,
 }) => {
+  const dispatch = useDispatch();
   const t = useI18nContext();
   const targetSubjectMetadata = useSelector((state) =>
     getTargetSubjectMetadata(state, snapId),
   );
 
   const snapName = getSnapName(snapId, targetSubjectMetadata);
 
-  if (isLoading) {
+  const interfaceData = useSelector((state) =>
+    getInterface(state, interfaceId),
+  );
+
+  const [internalState, setInternalState] = useState(
+    interfaceData?.interfaceState ?? {},
+  );
+
+  const snapRequestDebounced = debounce(
+    (params) =>
+      handleSnapRequest({
+        snapId,
+        origin: '',
+        handler: 'onUserInput',
+        request: {
+          jsonrpc: '2.0',
+          method: ' ',
+          params: {
+            event: params.value
+              ? {
+                  type: params.eventType,
+                  name: params.componentName,
+                  value: params.value,
+                }
+              : { type: params.eventType, name: params.componentName },
+            id: interfaceId,
+          },
+        },
+      }),
+    200,
+  );
+
+  const updateStateDebounced = debounce(
+    (state) => dispatch(updateInterfaceState(interfaceId, state)),
+    200,
+  );
+
+  const handleEvent = ({ eventType, componentName, parentForm, value }) => {
+    if (eventType) {
```

Comment:
This needs a comment to explain what is going on.

Also couldn't below be an else?